code/game/objects/objs.dm

```diff
@@ -97,7 +97,7 @@ var/global/list/reagents_to_log = list(FUEL, PLASMA, PACID, SACID, AMUTATIONTOXI
 			install_pai(W)
 			state_controls_pai(W)
 			playsound(src, 'sound/misc/cartridge_in.ogg', 25)
-	if(W)
+	if(istype(W))
```

Comment:
I feel like this isn't the first time this was tried, and I like it as much as I liked it the other times: not very.
Proper fix is to not abuse `attackby()`.
Hiding this runtime is just hiding shitcode.

Reply:
cowabunga it is